giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/core/service/processing/impl/ACompletedExtractionProcessor.java

```diff
@@ -99,4 +105,50 @@ protected IFile createFile(IFile file, IDocument document, String contentType, l
         
         return pagefile;
     }
+    
+    /**
+     * 
+     * This method will return a file if its already present in case of reprocessing else it will create a new file
+     * @param ipage, document, file, contentType, size, fileName, requestPrefix, function to find file of particular type
+     * @return existing file or newly created file
+     */
+    protected IFile getFile(IPage documentPage, IDocument document, IFile file, String contentType, long size, String fileName, String requestPrefix, Function<IPage, String> getFileId) {
+        IFile requestedFile;
+        if(documentPage != null && getFileId.apply(documentPage) != null && !getFileId.apply(documentPage).isEmpty()) {
+            requestedFile = filesService.getFileById(getFileId.apply(documentPage));
+        } else {
+            requestedFile = createFile(file, document, contentType, size, fileName, requestPrefix);
+            try {
+                filesService.saveFile(requestedFile);
+            } catch (UnstorableObjectException e) {
+                // should never happen, we're setting the id
+                messageHandler.handleMessage("Could not store file.", e, MessageType.ERROR);
+            }
+        }    
+        return requestedFile;
+    }
+    /**
+     * 
+     * This method will return a file if its already present in case of reprocessing else it will create a new file
+     * @param file, document, contentType, size, fileName, requestPrefix
+     * @return existing file or newly created file
+     */
+    protected IFile getFile(IFile file, IDocument document, String contentType, long size, String fileName, String requestPrefix) {
+        IFile completeText;
+        if (document.getExtractedTextFileId() != null && !document.getExtractedTextFileId().isEmpty()) {
```

Comment:
what about extracted images etc.? getting the id should probably handled via a passed function as well.

Reply:
In the CompletedTextExtractionProcessor we initially get the Extracted Text File Id of the document. We dont have any IPage here. 
Post this e process each of the requests pages for which we use the getFile method above. 

I will handle getExtractedTextFileId also as a function